server/src/main/java/org/opensearch/common/document/DocumentField.java

```diff
@@ -75,6 +90,110 @@ public DocumentField(String name, List<Object> values) {
         this.values = Objects.requireNonNull(values, "values must not be null");
     }
 
+    public DocumentField(byte[] in) throws IOException {
+        assert FeatureFlags.isEnabled(FeatureFlags.PROTOBUF) : "protobuf feature flag is not enabled";
```

Comment:
Also, please keep in mind that this classes (DocumentField, SearchHit, SearchHits, ....) are leaking through `RestClient` / `RestHighLevelClient`:  the feature flags are not relevant to the clients, those are only relevant to the server deployment.

Reply:
Also we should never use `assert` for feature flags in my opinion. Asserts are no-ops unless enabled when launching the JVM, so using it risks leaking feature flagged behavior to scenarios where it has not been enabled.